File: M3Tools/Types/Events/ProductEvents.vb
Namespace Events.Inventory
	Public Delegate Sub InventoryEventHandler(sender As Object, e As InventoryEventArgs)

	Public Class InventoryEventArgs
		Inherits BaseArgs

		Public Property Product As Types.Product

		Public Sub New(product As Types.Product, Optional eventType As EventType = Nothing)
			Me.Product = product
			Me.EventType = eventType
		End Sub
	End Class
End Namespace

File: M3Tools/Types/Events/ProductEvents.cs

namespace SPPBC.M3Tools.Events.Inventory
{
    public delegate void InventoryEventHandler(object sender, InventoryEventArgs e);

    public class InventoryEventArgs : BaseArgs
    {

        public Types.Product Product { get; set; }

        public InventoryEventArgs(Types.Product product, EventType eventType = default)
        {
            Product = product;
            EventType = eventType;
        }
    }
}